api/v1alpha1/tunnel_types.go

```diff
@@ -0,0 +1,77 @@
+/*
+MIT License
+
+Copyright (c) 2022 ngrok, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TunnelSpec defines the desired state of Tunnel
+type TunnelSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// ForwardsTo is the name and port of the service to forward traffic to
+	ForwardsTo string `json:"forwardsTo,omitempty"`
+
+	// Labels are key/value pairs that are attached to the tunnel
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+// TunnelStatus defines the observed state of Tunnel
```

Comment:
do you want to put a note here or log an issue with a little info about why we can't have a status here and how we could split it into separate controllers? I'm assuming we are putting that on the back burner but it would be good to not forget about it. I can spin up an issue with my understanding for you to review if you want?

Reply:
Yeah, I think we can have the status in here, but I saved it as a future enhancement. If you want to spin up an issue for that, I can review and add any additional context.